app/backend/src/core/controllers/appraiser/handlers/BlockRewardsHandler.ts

```diff
@@ -86,4 +88,49 @@ export class BlockRewardsHandler {
       blockRewardUsd: tipsUsd,
     };
   }
+
+  calculateBlockRewardByChainId(
+    gasUsed: bigint,
+    gasPrice: bigint,
+    baseFeePerGas: bigint,
+    priorityFeePerGas: bigint,
+    maxFeePerGas: bigint,
+  ): { gasFees: bigint; tips: bigint } {
+    const effectiveGasPrice =
+      baseFeePerGas + priorityFeePerGas != BigInt(0)
+        ? baseFeePerGas + priorityFeePerGas
+        : gasPrice;
+
+    let gasFees: bigint;
+    let tips: bigint;
+
+    switch (this.chainId) {
+      case chains.Optimism.chainId:
+        // Base fee isn't burned in OP
+        gasFees = gasUsed * effectiveGasPrice;
+        tips = gasUsed * effectiveGasPrice;
+        break;
+
+      case chains.zkEVM.chainId:
+        // maxFeePerGas, priorityFeePerGas and baseFeePerGas are null in zkEVM
+        gasFees = gasUsed * gasPrice;
+        tips = gasUsed * gasPrice;
+        break;
+
+      case chains.Starknet.chainId:
+        // TODO: Implement this
+        gasFees = BigInt(-1);
+        tips = BigInt(-1);
+        break;
+
+      default:
+        gasFees = BigInt(-1234);
```

Comment:
Maybe it would be better to throw or log and error instead of returning random values?